### PR TITLE
feat(storybook-dark-mode): add the storybook-dark-mode addon and StoryBookThemeProvider

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
…t -storybook-dark-mode

## Description

add storybook-dark-mode to centreon-ui
change the User attribute from theme to themeMode
have 2 ThemeProvider
- ThemeProvider that use userAtom to the themeMode and change the theme mode for centreon
- StoryBookThemeProvider that change the theme for StoryBook

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

see if the changed attribute of The user doesn't cause problem in outher files
see if centreon call the right ThemeProvider( ThemeProvider ot StoryBookThemeProvider)
see if storybook-dark-mode added correctly

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: packages/centreon-ui/src/index.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export { default as Panel } from './Panel';
 export { default as WithPanel } from './Panel/WithPanel';
 export { default as MemoizedPanel } from './Panel/Memoized';
 
-export { default as ThemeProvider } from './ThemeProvider';
+export { default as ThemeProvider } from './StoryBookThemeProvider';
```
Comment (bdauria) [inline packages/centreon-ui/src/index.ts:48]: The productive ThemeProvider should be exported here, not the Storybook one

File: packages/centreon-ui/src/testRenderer.tsx [hunk 2]
```diff
@@ -6,14 +6,17 @@ import {
...
 
 interface Props {
   children: React.ReactChild;
+  themeMode: ThemeMode;
 }
 
-const ThemeProviderWrapper = ({ children }: Props): JSX.Element => {
-  return <ThemeProvider>{children}</ThemeProvider>;
+const ThemeProviderWrapper = ({ children, themeMode }: Props): JSX.Element => {
+  return <ThemeProvider themeMode={themeMode}>{children}</ThemeProvider>;
```
Comment (bdauria) [inline packages/centreon-ui/src/testRenderer.tsx:19]: As this is used for test purposes only, the themeMode should be set to ThemeMode.light and not passed as a prop

**Review by Thebarda — APPROVED** [review]

File: packages/centreon-ui/src/Module/index.tsx [hunk 3]
```diff
@@ -4,27 +4,31 @@ import { Provider as JotaiProvider } from 'jotai';
...
+  themeMode,
 }: ModuleProps): JSX.Element => {
   const generateClassName = createGenerateClassName({
     seed: seedName,
   });
 
   return (
     <StylesProvider generateClassName={generateClassName}>
-      <ThemeProvider>
+      <ThemeProvider themeMode={themeMode}>
```
Comment (bdauria) [inline packages/centreon-ui/src/Module/index.tsx:31]: the themeMode should'nt be passed as a prop here, as it should be retrieved from the user atom within the ThemeProvider directly

File: packages/centreon-ui/src/StoryBookThemeProvider/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+declare module '@mui/material/styles/createPalette' {
+  interface TypeAction {
+    acknowledged: string;
+    acknowledgedBackground: string;
+    inDowntime: string;
+    inDowntimeBackground: string;
+  }
+}
```
Comment (bdauria) [inline packages/centreon-ui/src/StoryBookThemeProvider/index.tsx:26]: these typing declarations shouldn't be duplicated

File: packages/centreon-ui/.storybook/preview.js [hunk 5]
```diff
@@ -2,10 +2,15 @@ import React from 'react';
...
+import { useDarkMode } from 'storybook-dark-mode';
+
+import { ThemeMode } from '@centreon/ui-context';
+
 import { ThemeProvider } from '../src';
 
+
 const withThemeProvider = (story) => (
-  <ThemeProvider>{story()}</ThemeProvider>
+  <ThemeProvider themeMode={useDarkMode() ? ThemeMode.dark : ThemeMode.light}>{story()}</ThemeProvider>
```
Comment (bdauria) [inline packages/centreon-ui/.storybook/preview.js:13]: The Storybook theme provider should be used here

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**